### PR TITLE
Additional PHP 8 fixes

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Comment by samdark** [comment]
@yiisoft/core-developers, @yiisoft/reviewers need fresh pair of eyes. Seems I've made a mistake with all the effort somewhere and can't see where it is :(

File: framework/db/Exception.php [hunk 1]
```diff
@@ -27,9 +27,9 @@ class Exception extends \yii\base\Exception
      * @param string $message PDO error message
      * @param array $errorInfo PDO error info
      * @param string $code PDO error code
-     * @param \Exception $previous The previous exception used for the exception chaining.
+     * @param \Throwable|\Exception $previous The previous exception used for the exception chaining.
      */
-    public function __construct($message, $errorInfo = [], $code = '', \Exception $previous = null)
+    public function __construct($message, $errorInfo = [], $code = '', $previous = null)
```
Comment (cebe) [inline framework/db/Exception.php:32]: why is this needed?

File: framework/db/Command.php [hunk 2]
```diff
@@ -264,6 +267,9 @@ public function prepare($forRead = null)
             $message = $e->getMessage() . "\nFailed to prepare SQL: $sql";
             $errorInfo = $e instanceof \PDOException ? $e->errorInfo : null;
             throw new Exception($message, $errorInfo, (int) $e->getCode(), $e);
+        } catch (\Throwable $e) {
+            $message = $e->getMessage() . "\nFailed to prepare SQL: $sql";
+            throw new Exception($message, null, (int) $e->getCode(), $e);
```
Comment (cebe) [inline framework/db/Command.php:272]: why is `$errorInfo` null?

File: framework/db/Exception.php (hunk 1 above)
Comment (samdark) [reply framework/db/Exception.php:32]: Because PHP 8 introduced \Error and other native error types. These aren't extended from \Exception but passed here.

File: framework/db/Command.php (hunk 2 above)
Comment (samdark) [reply framework/db/Command.php:272]: Because `\Throwable` there is not an instance of `\PDOException`. Thus, nowhere to get `errorInfo` from.

File: tests/framework/db/ActiveRecordTest.php [hunk 3]
```diff
@@ -1893,24 +1893,24 @@ public function testLegalValuesForFindByCondition($modelClassName, $validFilter)
     public function illegalValuesForFindByCondition()
     {
         return [
-            [Customer::className(), ['id' => ['`id`=`id` and 1' => 1]]],
```
Comment (bizley) [inline tests/framework/db/ActiveRecordTest.php:1896]: Associative arrays are not allowed anymore in PHP 8 for `ReflectionMethod::invokeArgs()` hence the change.

**Review by bizley — APPROVED** [review]

File: tests/framework/db/ActiveRecordTest.php (hunk 3 above)
Comment (samdark) [reply tests/framework/db/ActiveRecordTest.php:1896]: Umm... maybe we can convert that somewhere in the middle instead of changing current behavior?

Comment (bizley) [reply tests/framework/db/ActiveRecordTest.php:1896]: But how is it changed? As far as I can see the output was using only array value anyway and that artificial keys (like `id`) were unneeded. Unless I don't know something here...

Comment (samdark) [reply tests/framework/db/ActiveRecordTest.php:1896]: Need to review it again. Anyway, big thanks for looking into it.
